apps/pie-storybook/stories/testing/pie-switch.test.stories.ts

```diff
@@ -0,0 +1,238 @@
+            },
+        },
+        label: {
+            description: 'The label text for the switch',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: 'Label',
+                },
+            },
+        },
+        labelPlacement: {
+            description: 'Set the placement of the label.',
+            control: 'select',
+            if: { arg: 'label', truthy: true },
+            options: labelPlacements,
+            defaultValue: {
+                summary: defaultProps.labelPlacement,
+            },
+        },
+        aria: {
+            description: 'The ARIA labels used for the switch.',
+            control: 'object',
+        },
+        name: {
+            description: 'Same as the HTML name attribute - indicates the name of the switch (for use with forms)',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: 'switch',
+                },
+            },
+        },
+        value: {
+            description: 'Same as the HTML value attribute - indicates the value of the switch (for use with forms). Defaults to "on".',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: defaultProps.value,
+                },
+            },
+        },
+        required: {
+            description: 'Same as the HTML required attribute - for use in forms',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.required,
+            },
+        },
+    },
+    args: defaultArgs,
+};
+
+export default switchStoryMeta;
+const changeAction = () => console.info('Switch clicked');
+
+const Template : TemplateFunction<SwitchProps> = (props) => {
+    const {
+        aria,
+        checked,
+        disabled,
+        label,
+        labelPlacement,
+        name,
+        value,
+        required,
+    } = props;
+
+    return html`
+        <pie-switch
+            id="pie-switch"
+            name="${name || nothing}"
+            value="${value || nothing}"
+            label="${label || nothing}"
+            labelPlacement="${label && labelPlacement ? labelPlacement : nothing}"
+            .aria="${aria}"
+            ?required="${required}"
+            ?checked="${checked}"
+            ?disabled="${disabled}"
+            @change="${changeAction}">
+        </pie-switch>`;
+};
+
+const FormTemplate: TemplateFunction<SwitchProps> = (props: SwitchProps) => html`
+    <p id="formLog" style="display: none; font-size: 2rem; color: var(--dt-color-support-positive);"></p>
+    <h2>Fake form</h2>
+    <form id="testForm">
+    <p>Required fields are followed by <strong><span aria-label="required">*</span></strong>.</p>
+
+    <section>
+    ${Template({
+    ...props,
+})}
+    </section>
+    <button id="submitButton" type="submit">Submit</button>
+    </form>
+    <script>
```

Comment:
please use a submit action like the change action above - there should be an example in text input I think. Ben did something similar for one of his migrations